lightning/src/routing/network_graph.rs

```diff
@@ -428,6 +428,70 @@ impl Readable for NodeInfo {
 	}
 }
 
+/// Allows for updating user's network metadata.
+pub trait UpdateMetadata {
+	/// Adds a failed route to track. If we see the route again, we'll reject it or we can take
+	/// some action (like set its fee to make route-finding not find that route again)
+	fn add_failed_route(&mut self, route: Vec<Vec<RouteHop>>);
```

Comment:
I dont think this needs the double-vec - we only care about one path of an MPP failing, not the full route.